proc/userintr.c

```diff
@@ -81,21 +79,17 @@ int userintr_setHandler(unsigned int n, int (*f)(unsigned int, void *), void *ar
 {
 	process_t *process = proc_current()->process;
 	userintr_t *ui;
-	cond_t *cond = NULL;
 	int id, res;
+	_Atomic(u32) *cond_futex = (_Atomic(u32) *)(addr_t)c;
 
 	if (c > 0) {
-		cond = cond_get(c);
-		if (cond == NULL) {
+		if (vm_mapBelongs(process, cond_futex, sizeof(*cond_futex)) < 0) {
```

Comment:
This check, along with `handle_t` conversion, should be moved into `syscall_interrupt`.